### PR TITLE
New rules: native_binary and native_test

### PR DESCRIPTION
**Pull request by laszlocsomor**
native_binary() wraps a pre-built binary or script
in a *_binary rule interface. Rules like genrule
can tool-depend on it, and it can be executed with
"bazel run". This rule can also augment the binary
with runfiles.

native_test() is similar, but creates a testable
rule instead of a binary rule.

Neither rule requires Bash.

Fixes https://github.com/bazelbuild/bazel-skylib/issues/148

RELNOTES[NEW]: The new `native_binary()` and `native_test()` rules let you wrap a pre-built binary in a binary and test rule respectively.

### CODE REVIEW

**Comment by laszlocsomor** [comment]
All green, sorry for the noise till now.

File: rules/native_binary.bzl [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+do, but they run the wrapped binary directly, instead of through Bash, so they
+don't depend on Bash and work with --shell_exectuable="".
+"""
+
+load("//rules/private:copy_file_private.bzl", "copy_bash", "copy_cmd")
+
+def _impl_rule(ctx, is_windows):
+    if not ctx.attr.out:
+        fail(msg = "Must not be empty", attr = "out")
+    out = ctx.actions.declare_file(ctx.attr.out)
```
[the remark this answers is not in the document]
Comment (laszlocsomor) [reply rules/native_binary.bzl]: Yes, but then it's not `select()`'able.
What do you recommend?

File: rules/native_binary.bzl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+These rules let you wrap a pre-built binary or script in a conventional binary
+and test rule respectively. They fulfill the same goal as sh_binary and sh_test
+do, but they run the wrapped binary directly, instead of through Bash, so they
+don't depend on Bash and work with --shell_exectuable="".
+"""
+
+load("//rules/private:copy_file_private.bzl", "copy_bash", "copy_cmd")
+
+def _impl_rule(ctx, is_windows):
+    if not ctx.attr.out:
```
[the remark this answers is not in the document]
Comment (laszlocsomor) [reply rules/native_binary.bzl]: Done.

File: rules/native_binary.bzl (hunk 1 above)
[the remark this answers is not in the document]
Comment (laurentlb) [reply rules/native_binary.bzl]: Good point. Your code looks fine then, thanks for the new comment.

[the remark this answers is not in the document]
Comment (laszlocsomor) [reply rules/native_binary.bzl]: Thanks!

**Pull request merged**